telethon/client/downloads.py

```diff
@@ -439,7 +439,9 @@ class DownloadMethods:
                     if inspect.isawaitable(r):
                         await r
 
-            f.flush()
+            if hasattr(f, 'flush') and callable(getattr(f, 'flush')):
```

Comment:
```suggestion
            if callable(getattr(f, 'flush', None)):
```

As a side note, there's often no need to call `getattr` with a string literal, because that's just `f.flush`.

Reply:
> Have you searched the code base for other usages of `flush`, or is this the only one?

This the only one.